src/Test.elm

```diff
@@ -0,0 +1,303 @@
+module Test exposing (Test, FuzzOptions, describe, test, filter, concat, fuzz, fuzz2, fuzz3, fuzz4, fuzz5, fuzzWith)
+
+{-| Writing tests.
+
+@docs Test, test
+
+## Organizing Tests
+
+@docs describe, concat, filter
+
+## Fuzz Testing
+
+@docs fuzz, fuzz2, fuzz3, fuzz4, fuzz5, fuzzWith, FuzzOptions
+-}
+
+import Random
+import Test.Internal as Internal
+import Expect exposing (Expectation)
+import Random exposing (Generator)
+import Fuzz exposing (Fuzzer)
+
+
+{-| A test which has yet to be evaluated. When evaluated, it produces one
+or more [`Expectation`](../Expect#Expectation)s.
+
+See [`test`](#test) and [`fuzz`](#fuzz) for some ways to create a `Test`.
+-}
+type alias Test =
+    Internal.Test
+
+
+{-| Run the given tests in order.
+
+    concat [ testDecoder, testSorting ]
+-}
```

Comment:
Yeah, I was debating whether this would be worth maintaining, and decided it would be for a consistent UX.

When I see a different failed test at the top of my failures list, I think "cool, fixed it, time to context-switch to the next failure." If it turned out this was just that things had finished in a different order, it sounds like it would lead to frustration pretty easily.

From an implementation standpoint this means we can't always report failures quite as soon as we know about them. I doubt the delays will be noticeable in practice, whereas I think inconsistent ordering would get annoying fast.